src/DictVectors/initiators.jl

```diff
@@ -0,0 +1,219 @@
+    threshold::V
+end
+
+function value(i::Initiator, v::InitiatorValue)
+    return v.safe + v.initiator + !iszero(v.initiator) * v.unsafe
+end
+
+"""
+    SimpleInitiator
+
+Simplified initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators cannot spawn.
+"""
+struct SimpleInitiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::SimpleInitiator, v::InitiatorValue)
+    return v.safe + v.initiator
+end
+
+"""
+    CoherentInitiator
+
+Initiator to be passed to [`InitiatorDVec`](@ref). Initiator rules:
+
+* Initiators can spawn anywhere.
+* Non-initiators can spawn to initiators.
+* Multiple non-initiators can spawn to a single non-initiator if their contributions add up
+  to a value greater than the initiator threshold.
+"""
+struct CoherentInitiator{V} <: AbstractInitiator{V}
+    threshold::V
+end
+
+function value(i::CoherentInitiator, v::InitiatorValue)
+    if !iszero(v.initiator) || abs(v.unsafe) > i.threshold
+        return v.initiator + v.safe + v.unsafe
+    else
+        return v.initiator + v.safe
+    end
+end
+
+"""
+    InitiatorDVec
+
+Functionally identical to [`DVec`](@ref), but contains [`InitiatorValue`]s
+"""
+struct InitiatorDVec{
+    K,V,D<:Dict{K,InitiatorValue{V}},S<:StochasticStyle{V},I<:AbstractInitiator
+} <: AbstractDVec{K,V}
+    storage::D
+    style::S
+    initiator::I
+end
+
+###
+### Constructors
+###
+function InitiatorDVec(
+    dict::Dict{K,V}; style=default_style(V), initiator=Initiator(one(V)), capacity=0
+) where {K,V}
+    T = eltype(style)
+    storage = Dict{K,InitiatorValue{T}}()
+    sizehint!(storage, capacity)
+    for (k, v) in pairs(dict)
+        storage[k] = InitiatorValue{T}(safe=v)
+    end
+    return InitiatorDVec(storage, style, initiator)
+end
+function InitiatorDVec(args...; kwargs...)
+    storage = Dict(args...)
+    return InitiatorDVec(storage; kwargs...)
+end
+function InitiatorDVec(
+    dv::AbstractDVec{K,V}; style=StochasticStyle(dv), initiator=Initiator(one(V)), capacity=0
+) where {K,V}
+    storage = Dict{K,InitiatorValue{V}}()
+    sizehint!(storage, max(capacity, length(dv)))
+    for (k, v) in pairs(dv)
+        storage[k] = InitiatorValue{V}(safe=v)
+    end
+    return InitiatorDVec(storage, style, initiator)
+end
+function InitiatorDVec{K,V}(; kwargs...) where {K,V}
+    return InitiatorDVec(Dict{K,V}(); kwargs...)
+end
+
+function Base.empty(dvec::InitiatorDVec)
+    return InitiatorDVec(empty(dvec.storage), dvec.style, dvec.initiator)
+end
+function Base.empty(dvec::InitiatorDVec, args...)
+    return InitiatorDVec(empty(dvec.storage, args...); initiator=dvec.initiator)
+end
+Base.similar(dvec::InitiatorDVec, args...; kwargs...) = empty(dvec, args...; kwargs...)
+
+###
+### Show
+###
+function Base.summary(io::IO, dvec::InitiatorDVec{K,V,S}) where {K,V,S}
+    len = length(dvec)
+    print(io, "InitiatorDVec{$K,$V,$S} with $len entries")
+end
```

Comment:
I think it currently does not work as intended. In particular it does not print stochasticstyle or initiator information, which would be quite helpful.
```suggestion
function Base.summary(io::IO, dvec::InitiatorDVec{K,V}) where {K,V}
    len = length(dvec)
    print(io, "InitiatorDVec{$K,$V} with $len entries, style = $(dvec.style), initiator = $(dvec.initiator)")
end
```

Reply:
I'll also do this for regular `DVec`.